### ISSUE TITLE
Error: Module build failed CssSyntaxError

### ISSUE THREAD

**Issue by cseas** [vanilla-extract-css/vanilla-extract#968]
### Describe the bug

Adding Vanilla Extract webpack plugin to a project with existing rules for processing css files throws this error.

The error goes away if we remove the existing rules for `test: /\.css$/i` but that shouldn't be required as we might still need to import css files in the project and we want webpack to process those.

Removing the below lines in the reproduction code fixes the error:
https://github.com/cseas/repro/blob/main/webpack.base.mjs#L32-L33

### Reproduction

https://github.com/cseas/repro

### System Info

```shell
System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 94.64 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.13.0 - /opt/homebrew/opt/node@18/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.19.3 - /opt/homebrew/opt/node@18/bin/npm
  Browsers:
    Chrome: 107.0.5304.87
    Firefox: 101.0.1
    Safari: 16.2
  npmPackages:
    @vanilla-extract/webpack-plugin: ^2.2.0 => 2.2.0 
    webpack: ^5.75.0 => 5.75.0
```

### Used Package Manager

npm

### Logs

```shell
ERROR in ./node_modules/@vanilla-extract/webpack-plugin/extracted.js (./node_modules/@vanilla-extract/webpack-plugin/extracted.js.webpack[javascript/auto]!=!./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[1].use[1]!./node_modules/style-loader/dist/cjs.js!./node_modules/css-loader/dist/cjs.js!./node_modules/@vanilla-extract/webpack-plugin/virtualFileLoader/dist/vanilla-extract-webpack-plugin-virtualFileLoader.cjs.js?{"fileName":"src/App.css.ts.vanilla.css","source":"Ll81MWRlaWUwIHsKICBwYWRkaW5nOiAxMHB4Owp9"}!./node_modules/@vanilla-extract/webpack-plugin/extracted.js)
Module build failed (from ./node_modules/css-loader/dist/cjs.js):
CssSyntaxError

(2:7) /Users/abhijeet.singh/Documents/cseas/repro/node_modules/@vanilla-extract/webpack-plugin/extracted.js Unknown word

  1 | 
> 2 |       import API from "!../../style-loader/dist/runtime/injectStylesIntoStyleTag.js";
    |       ^
  3 |       import domAPI from "!../../style-loader/dist/runtime/styleDomAPI.js";
  4 |       import insertFn from "!../../style-loader/dist/runtime/insertBySelector.js";
 @ src/App.css.ts.vanilla.css!=!./node_modules/@vanilla-extract/webpack-plugin/virtualFileLoader/dist/vanilla-extract-webpack-plugin-virtualFileLoader.cjs.js?{"fileName":"src/App.css.ts.vanilla.css","source":"Ll81MWRlaWUwIHsKICBwYWRkaW5nOiAxMHB4Owp9"}!./node_modules/@vanilla-extract/webpack-plugin/extracted.js
 @ ./src/App.css.ts 1:0-327
 @ ./src/App.tsx 2:0-41 4:39-48
 @ ./src/index.tsx 4:0-31 6:46-49
```

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by askoufis** [comment]
You could change the existing CSS rule to exclude `.vanilla.css` files:
```diff
-test: /\.css$/i
+test: /(?<!\.vanilla)\.css$/i
```

Does that work for your use case?

**Comment by cseas** [comment]
@askoufis Thank you! That solution works. However, I personally find that regex a bit unreadable. Wasn't even aware about negative lookbehind until I saw your snippet.

I stumbled into [this comment][1] that mentions we can achieve the same result with the exclude option. So all I had to do was put the regex for vanilla extract in exclude. This fixed it:

```js
      {
        test: /\.css$/i,
        use: ["style-loader", "css-loader"],
        exclude: /\.vanilla\.css$/i,
      }
```

This does look like something that should be mentioned in the Vanilla Extract docs, since having an existing rule for css processing isn't a rare use-case.

[1]: https://github.com/webpack-contrib/css-loader/issues/672#issuecomment-362480828
